tools/src/versioning/android/versionCxx.ts

```diff
@@ -0,0 +1,158 @@
+import assert from 'assert';
+import fs from 'fs-extra';
+import glob from 'glob-promise';
+import path from 'path';
+import spawnAsync from '@expo/spawn-async';
+
+import { ANDROID_DIR, PACKAGES_DIR } from '../../Constants';
+import { transformFileAsync } from '../../Transforms';
+
+export async function versionCxxExpoModulesAsync(version: string) {
+  await new VersionerExpoAv(version).process();
+}
```

Comment:
Why to create a class?
Will it be used in some other places too?
Or the internal state required for this pipeline to run smoothly needs some class-based approach? 🤔 
I see it as simple pipeline that can be written like this:
```suggestion
const PACKAGES_TO_VERSION = [
  'expo-av',
];

export async function versionCxxExpoModulesAsync(version: string) {
  const abiName = `abi${version.replace(/\./g, '_')}`;

  try {
    for (const package of PACKAGES_TO_VERSION) {
      const packageRoot = path.join(PACKAGES_DIR, package, 'android');
      const versionedAbiRoot = path.join(
        ANDROID_DIR,
        'versioned-abis',
        `expoview-${this.abiName}`,
        'src',
        'main'
      );
  
      await Promise.all([
        patchCMake(packageRoot, abiName),
        patchGradle(packageRoot, abiName),
        patchCxxFilesForJavaDescriptor(packageRoot, abiName),
      ]);
      await spawnGradleCopyReleaseJniLibsCommand();
      await Promise.all([
        copySoPrebuiltLibs(versionedAbiRoot),
        rewriteSystemLoadLibs(versionedAbiRoot),
      ]);
    }
  } catch (e) {
    throw e;
  } finally {
    await this.resetPatches(...);
  }
}
```